src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -723,6 +749,18 @@ private static void validateMinRFThreshold(int warn, int fail, String name)
         validateMinRFVersusDefaultRF(fail, name);
     }
 
+    private static void validateMaxRFThreshold(int warn, int fail, String name)
+    {
+        validateMaxIntThreshold(warn, fail, name);
+
+        if (fail != -1 && fail < DatabaseDescriptor.getDefaultKeyspaceRF())
+        {
+            throw new IllegalArgumentException(format("maximum_replication_factor_fail_threshold to be set (%d) " +
+                                                      "cannot be lesser than default_keyspace_rf (%d)",
+                                                      fail, DatabaseDescriptor.getDefaultKeyspaceRF()));
+        }
```

Comment:
This is checked at startup and when we dynamically set a new `maximum_replication_factor_fail_threshold`. However, we are missing an equivalent check for dynamic updates of `default_keyspace_rf`, as it's done for max RF in [`DatabaseDescriptor.setDefaultKeyspaceRF(int)`](https://github.com/apache/cassandra/blob/1b3aec8eef6d1348177dc5a1f0ef30d4ab8b79ee/src/java/org/apache/cassandra/config/DatabaseDescriptor.java#L4087-L4090).